Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/CollectionViews/CollectionView.cs

```diff
@@ -0,0 +1,1281 @@
+        /// (This method is used by a listener's (e.g. System.Windows.Controls.ItemsControl)
+        /// CollectionChanged event handler to speed up its reaction to insertion and deletion of items.
+        /// If IndexOf is  not implemented, a listener does a binary search using IComparer.Compare.)
+        /// </remarks>
+        /// <param name="item">data item</param>
+        /// <returns>The index where the given item belongs, or -1 if this index is unknown.</returns>
+        public abstract int IndexOf(object item);
+
+        /// <summary>
+        /// Retrieve item at the given zero-based index in this CollectionView.
+        /// </summary>
+        /// <remarks>
+        /// <p>The index is evaluated with any SortDescriptions or Filter being set on this CollectionView.
+        /// If the underlying collection is only of type IEnumerable, this method
+        /// is a O(N) operation.</p>
+        /// <p>When deriving from CollectionView, override this method to provide
+        /// a more efficient implementation.</p>
+        /// </remarks>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown if index is out of range
+        /// </exception>
+        /// <returns>Item at the given zero-based index in this CollectionView.</returns>
+        public abstract object GetItemAt(int index);
+
+        /// <summary>
+        /// Gets the number of items (or -1, meaning "don't know");
+        /// if a Filter is set, this counts only items that pass the filter.
+        /// </summary>
+        /// <remarks>
+        /// <p>If the underlying collection is only of type IEnumerable, this count
+        /// is a O(N) operation; this Count value will be cached until the
+        /// collection changes again.</p>
+        /// <p>When deriving from CollectionView, override this property to provide
+        /// a more efficient implementation.</p>
+        /// </remarks>
+        public abstract int Count
+        {
+            get;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the resulting (filtered) view is emtpy.
+        /// </summary>
+        public abstract bool IsEmpty
+        {
+            get;
+        }
+
+        /// <summary>
+        /// Gets an object that compares items in this view.
+        /// </summary>
+        public virtual IComparer Comparer
+        {
+            get
+            {
+                return this as IComparer;
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether this view needs to be refreshed.
+        /// </summary>
+        public virtual bool NeedsRefresh
+        {
+            get
+            {
+                return CheckFlag(CollectionViewFlags.NeedsRefresh);
+            }
+        }
+
+        /// <summary>
+        /// Raise this event when the (filtered) view changes
+        /// </summary>
+        protected virtual event NotifyCollectionChangedEventHandler CollectionChanged;
```

Comment:
Is this event (And the Interface event redirects above) necessary if the Callbacks are overridable?

Reply:
I'd rather keep this code as is instead of refactoring it at this point. Maybe some refactoring can be done in the future, but I would give it low priority.